src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java

```diff
@@ -155,11 +157,27 @@ private static Query getFilterQuery(CreateQueryRequest createQueryRequest) {
                     createQueryRequest.k
                 )
             );
+            final Query filterQuery;
             try {
-                return createQueryRequest.getFilter().get().toQuery(queryShardContext);
+                filterQuery = createQueryRequest.getFilter().get().toQuery(queryShardContext);
             } catch (IOException e) {
                 throw new RuntimeException("Cannot create knn query with filter", e);
             }
+            // If k-NN Field is nested field then parentFilter will not be null. This parentFilter is set by the
```

Comment:
Does that mean, we can use filter only on either nested field or top level fields but not both?

Reply:
No thats not the case. You can use on both as long as that filter is giving some ids.

Reply:
But ideally its suited to use filters with one type of field, in my recommendation. I added a test for this specific case